subsys/net/l2/canbus/canbus.c

```diff
@@ -1096,9 +1490,14 @@ static enum net_verdict canbus_recv(struct net_if *iface,
 
 		if (pkt->can_rx_ctx->state == NET_CAN_RX_STATE_FIN) {
 			rx_finish(pkt);
+
+#ifdef CONFIG_NET_L2_CANBUS_ETH_TRANSLATOR
```

Comment:
Change this to use IS_ENABLED(CONFIG_NET_L2_CANBUS_ETH_TRANSLATOR) instead so that we get the compiler to check both branches.